pkg/apis/dynatrace/v1alpha1/defaults.go

```diff
@@ -13,4 +20,28 @@ func SetDefaults_OneAgentSpec(obj *OneAgentSpec) {
 	if len(obj.Args) == 0 {
 		obj.Args = append(obj.Args, "APP_LOG_CONTENT_ACCESS=1")
 	}
+
+	if _, ok := obj.NodeSelector["beta.kubernetes.io/os"]; !ok {
```

Comment:
Is this sufficient? What if the user specifies multiple node selectors, does this still work as desired? So, is `NodeSelector` as map?

Reply:
`NodeSelector` is of type `map[string]string`. Other selectors do not get overwritten.